3party/opening_hours/opening_hours.hpp

```diff
@@ -0,0 +1,566 @@
+/*
+  The MIT License (MIT)
+
+  Copyright (c) 2015 Mail.Ru Group
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <iostream>
+#include <chrono>
+#include <type_traits>
+
+namespace osmoh
+{
+
+class Time
+{
+  enum State
+  {
+    IsNotTime = 0,
+    HaveHours = 1,
+    HaveMinutes = 2,
+  };
+
+  using TStateRep = std::underlying_type<State>::type;
+
+public:
+  enum class Event
+  {
+    NotEvent,
+    Dawn,
+    Sunrise,
+    Sunset,
+    Dusk
+  };
+
+  using THours = std::chrono::hours;
+  using TMinutes = std::chrono::minutes;
+
+  Time() = default;
+  Time(Time const &) = default;
+  Time(THours const hours);
```

Comment:
Are you talking about THours?
It's not necessary, cause underlying type of chrono::hours is integral.